src/entity/EnemyShipFormation.java

```diff
@@ -558,53 +577,105 @@ public final int[] _destroy(final Bullet bullet, final EnemyShip destroyedShip,
 	 * 			  explosive EnemyShip's x-coordinates in EnemyShips
 	 * @param index_y
 	 * 			  explosive EnemyShip's y-coordinates in EnemyShips
+	 * @param enemyShips
+	 * 			  the current arrangement of the enemy
 	 */
+	public int[] explosive(final int x, final int y, final int index_x, final int index_y, List<List<EnemyShip>> enemyShips){
 
-	public void explosive(final int x, final int y,final int index_x, final int index_y) {
-		javax.swing.Timer timer = new javax.swing.Timer(200, null);
-		final int[] i = {1};
-
+		Queue<EnemyShip> targetShipQ = new LinkedList<>();
+		Timer timer = new Timer(500, null);
+		int range = 2;
+		int i = 1;
+		int point = 0;
+		int mob = 0;
 
 		Bullet bullet = new Bullet(0,0,-1);
-		ActionListener listener = new ActionListener() {
-			@Override
-			public void actionPerformed(ActionEvent e) {
 
+		do{
+
+			if(enemyShips.size() > index_x+i && enemyShips.get(index_x+i).size() > y){ // right
```

Comment:
I added checking index code and that is in 'develop' branch. I'll add that here code when I resolve conflict on this PR!